service-wfs-client/src/main/java/org/oskari/service/wfs/client/OskariWFS110Client.java

```diff
@@ -46,58 +47,57 @@ private OskariWFS110Client() {}
     public static SimpleFeatureCollection getFeatures(String endPoint, String user, String pass,
             String typeName, ReferencedEnvelope bbox, CoordinateReferenceSystem crs, int maxFeatures, Filter filter) {
         // First try GeoJSON
+        Map<String, String> query = getQueryParams(typeName, bbox, crs, maxFeatures, filter);
+        query.put("OUTPUTFORMAT", "application/json");
+
+        byte[] response = getResponse(endPoint, user, pass, query);
         try {
-            return getFeaturesGeoJSON(endPoint, user, pass, typeName, bbox, crs, maxFeatures, filter);
+            return parseGeoJSON(new ByteArrayInputStream(response), crs);
         } catch (Exception e) {
-            LOG.warn(e, "Failed to read WFS response as GeoJSON");
+            if (!isOutputFormatInvalid(new ByteArrayInputStream(response))) {
+                // If we can not determine that the exception was due to bad
+                // outputFormat parameter then don't bother trying GML
+                throw new ServiceRuntimeException("Unable to get features");
+            }
+            LOG.debug(e, "Failed to parse GeoJSON response");
```

Comment:
Move logging to be done before the exception throw to get the cause.

Reply:
Actually the logging should be done only inside the if block (definitely not after as it is now). If we can determine that the error was because the response was a valid XML <ows:ExceptionReport> stating that the error is because of the `&outputFormat=application/json` then we don't need to log that exception.

Reply:
Decided to just not log anything here, but instead attach the cause to the ServiceRuntimeException if one is thrown